third_party/java/arcs/build_defs/internal/schemas.bzl

```diff
@@ -112,50 +114,43 @@ def arcs_kt_schema(
         for ext in platforms:
             if ext not in supported:
                 fail("Platform %s not allowed; only %s supported.".format(ext, supported.join(",")))
-            wasm = ext == "wasm"
             genrule_name = replace_arcs_suffix(src, "_genrule_" + ext)
-            out = replace_arcs_suffix(src, "_GeneratedSchemas.%s.kt" % ext)
-            outs.append(out)
-            _run_schema2wasm(
+            outs.append(genrule_name)
+            schema2pkg(
                 name = genrule_name,
-                src = src,
-                out = out,
-                deps = deps + data,
-                wasm = wasm,
-                language_flag = "--kotlin",
-                language_name = "Kotlin",
+                srcs = [src],
+                platform = ext,
+                data = data,
+                testonly=testonly,
             )
 
     arcs_kt_library(
         name = name,
-        srcs = outs,
+        srcs = [":" + out for out in outs],
         platforms = platforms,
-        deps = ARCS_SDK_DEPS,
+        deps = ARCS_SDK_DEPS + deps,
         visibility = visibility,
+        testonly=testonly
     )
     outdeps = outdeps + ARCS_SDK_DEPS
 
-    if (test_harness):
+    if test_harness:
         test_harness_outs = []
         for src in srcs:
-            out = replace_arcs_suffix(src, "_TestHarness.kt")
+            out = replace_arcs_suffix(src, "_genrule_test_harness")
             test_harness_outs.append(out)
 
-            _run_schema2wasm(
```

Comment:
Can you remove the definition of _run_schema2wasm now?

Reply:
This was a good reminder: I should change the cpp side of the rules (that's the last place where this fn is used).